pool/pool.go

```diff
@@ -199,6 +210,12 @@ func (p *Pool) Get() (*redis.Client, error) {
 	case <-p.stopCh:
 		return nil, errors.New("pool emptied")
 	default:
+		var timeoutCh <-chan time.Time
+		if p.po.getTimeout > 0 {
+			timeoutCh = time.After(p.po.getTimeout)
```

Comment:
```
timer := time.NewTimer(p.po.getTimeout)
defer timer.Stop()
timeoutCh = timer.C
```
Is be better, that way there's not a hanging go-routine after every call to Get which won't get cleaned up (similar to how you have to cancel timeout contexts even if they're not canceled).